static/app/views/dashboards/widgets/tableWidget/tableWidgetVisualization.tsx

```diff
@@ -139,18 +153,59 @@ export function TableWidgetVisualization(props: TableWidgetVisualizationProps) {
     };
   };
 
+  const {data, meta} = tableData;
+  const locationSort = decodeSorts(location?.query?.sort)[0];
+
+  let widths = new Array(columns?.length ?? Object.keys(meta.fields).length).fill(
+    COL_WIDTH_UNDEFINED
+  );
+  const locationWidths = location.query?.width;
+  // If at least one column has the width key and that key is defined, take that over url widths
+  if (columns?.some(column => defined(column.width))) {
+    widths = columns.map(column =>
+      defined(column.width) ? column.width : COL_WIDTH_UNDEFINED
+    );
+  } else if (resizable && Array.isArray(locationWidths)) {
```

Comment:
Maybe also good to check that `locationWidths` has the same number of elements as the number of columns?